gentzkow/template
# PR for #71 Log software + packages versions

This pull request closes https://github.com/gslab-econ/gslab_make/issues/56 and https://github.com/gentzkow/template/issues/71. This template branch is currently pulling the 56-version-log branch of the gslab_make submodule, so https://github.com/gslab-econ/gslab_make/pull/56 should be closed first.

The steps to test the new feature are the following:
- Clone template and switch to the issue71-log-version branch
- Create the template conda environment (i.e., run conda env create -f setup/conda_env.yaml from root)
- Load the submodule by running:
```
git submodule init
git submodule update
```
- Navigate to lib/gslab_make and run `git switch 56-version-log ` (we are testing the 56-version-log branch of gslab_make). After that, run git branch. The current branch of the submodule should be 56-version-log.
- Go back to root and activate the template environment. Run check_setup.py from the setup module.
- Run run_all.py from root. Check the `make.log` and `version.log` files in each module.

Once the tests are successful, we can proceed to merge the 56-version-log branch of gslab_make to master, point the submodule to master and run the entire repo again.

I'm adding @snairdesai as a reviewer. 

cc. @gentzkow
